### ISSUE TITLE
Improve complexity of run algorithm

### ISSUE THREAD

**Issue by merelcht** [kedro-org/kedro#4373]
## Description
The `_run` method was unified in #4353. However, complexity wasn't really taken into account during the refactoring. The complexity of the `_run` method should be revisited and improved if possible. 

## Context
Original comment by @DimedS: https://github.com/kedro-org/kedro/pull/4353#discussion_r1878275022

**Comment by merelcht** [comment]
While working on https://github.com/kedro-org/kedro/pull/4502, I had a look at this issue again too. With the changes in the PR for `SequentialRunner` it again simply executes an ordered list of nodes sequentially, so I don't think https://github.com/kedro-org/kedro/pull/4353#discussion_r1878275022 still holds. What do you think @DimedS ?

**Comment by DimedS** [comment]
You're right, that problem no longer exists for `SequentialRunner`, so we can close this issue.
